### PR TITLE
[GEOS-9502] <VendorOption> in <LineSymbolizer> validation bug

### PR DESCRIPTION
**Pull request by imranrajjad**
JIRA : https://osgeo-org.atlassian.net/browse/GEOS-9502

-updated
/gs-main/src/main/resources/schemas/sld/StyledLayerDescriptor.xsd
-added unit test

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Committs changing the REST API, or any configuration object, should check it the REST API docs (Swagger YAML files and classic documentation) need to be updated.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by imranrajjad** [comment]
@aaime Java8 unrelated failure on GWC module, can you please re-kick? thank you

**Comment by aaime** [comment]
Rekicked

File: src/main/src/main/resources/schemas/sld/StyledLayerDescriptor.xsd [hunk 1]
```diff
@@ -359,6 +359,7 @@
             <xsd:element ref="sld:Geometry" minOccurs="0"/>
             <xsd:element ref="sld:Stroke" minOccurs="0"/>
             <xsd:element ref="sld:PerpendicularOffset" minOccurs="0"/>
+            <xsd:element ref="sld:VendorOption" minOccurs="0" maxOccurs="unbounded" />
```
Comment (aaime) [inline src/main/src/main/resources/schemas/sld/StyledLayerDescriptor.xsd]: Change here looks good, what happens however if SLD 1.1 is used?

**Comment by imranrajjad** [comment]
@aaime unit test added for 1.1.0 also

**Comment by aaime** [comment]
Looks good thanks. I see there is a conflict.

**Comment by imranrajjad** [comment]
@aaime will resolve and ping back

**Comment by imranrajjad** [comment]
@aaime conflicts resolved

**Comment by nmco** [comment]
I restarted the Windows build.

**Comment by nmco** [comment]
The build failure looks unrelated (I already restarted):
image

@imranrajjad I'm going to merge, but please keep an eye on GeoServer builds.

**Pull request merged**